### PR TITLE
allow avatar walk-speed to be set from javascript

### PR DESCRIPTION
**Pull request by sethalves**
- allow avatar walk-speed to be set from javascript

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6028/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6028/HighFidelity-Beta-PR12099-3cd0ec3336618235188002ee348d9ff16c0e6aa2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6028/HighFidelity-Beta-PR12099-3cd0ec3336618235188002ee348d9ff16c0e6aa2.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6079/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6079/HighFidelity-Beta-PR12099-1c94c9c11dd52abd4d30976f8bd7ba40720e6722.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6079/HighFidelity-Beta-PR12099-1c94c9c11dd52abd4d30976f8bd7ba40720e6722.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6100/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6100/HighFidelity-Beta-PR12099-ad389e5072efb2be5961bc014f6b99143bb47b16.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6105/HighFidelity-Beta-PR12099-6626f556520e0e16f2331466aa3b121c24b52cf2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6105/HighFidelity-Beta-PR12099-6626f556520e0e16f2331466aa3b121c24b52cf2.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6105/app_12099_PR.apk

File: scripts/system/run.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"use strict";
+
+/* global Script, Tablet, MyAvatar */
+
+(function() { // BEGIN LOCAL_SCOPE
+    var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
+    var button = tablet.addButton({
+        icon: Script.resolvePath("assets/images/icon-particles.svg"),
```
Comment (howard-stearns) [inline scripts/system/run.js]: icon-particles???

Comment (sethalves) [reply scripts/system/run.js]: it seemed closest

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6111/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6111/HighFidelity-Beta-PR12099-41785e8a88f964833a0352cfa77f7a7739f86806.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6111/HighFidelity-Beta-PR12099-41785e8a88f964833a0352cfa77f7a7739f86806.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6139/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6139/HighFidelity-Beta-PR12099-72ac1dd216f13a827b82fec20e636cfb29ea1b70.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6139/HighFidelity-Beta-PR12099-72ac1dd216f13a827b82fec20e636cfb29ea1b70.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6170/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6170/HighFidelity-Beta-PR12099-3816c732c0411aff9df647db09256a8347a9f0e2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6170/HighFidelity-Beta-PR12099-3816c732c0411aff9df647db09256a8347a9f0e2.dmg

File: scripts/system/run.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        text: "Run",
+        sortOrder: 15
+    });
+
+    function onClicked() {
+        if (MyAvatar.walkSpeed < 4) {
+            MyAvatar.walkSpeed = 4.5;
```
Comment (AndrewMeadows) [inline scripts/system/run.js]: If less than `4` set to `4.5`?  If we ever hit the else case below then `MyAvatar.walkSpeed` will be `2.6`.  On subsequent click (`2.6` is indeed less than `4`) and `MyAvatar.walkspeed` will be set to `4.5`.

If that is by design... it doesn't make sense to me.

Comment (sethalves) [reply scripts/system/run.js]: it's a toggle.  if you are walking when you press the button, you'll run.  if you are running, you'll walk.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6227/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6227/HighFidelity-Beta-PR12099-57368fb786f68e7e728c6efa009056fb321bee7b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6227/HighFidelity-Beta-PR12099-57368fb786f68e7e728c6efa009056fb321bee7b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6268/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6268/HighFidelity-Beta-PR12099-f6ce25405929a68acc8c5f3b550e94128350cdf3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6268/HighFidelity-Beta-PR12099-f6ce25405929a68acc8c5f3b550e94128350cdf3.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6297/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6297/HighFidelity-Beta-PR12099-eaeebc02153516fe6c0d807dbf4bc4becd73190a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6297/HighFidelity-Beta-PR12099-eaeebc02153516fe6c0d807dbf4bc4becd73190a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6320/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6320/HighFidelity-Beta-PR12099-12db51728ecff8673aebc3accc61a9ec72a1bed9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6320/HighFidelity-Beta-PR12099-12db51728ecff8673aebc3accc61a9ec72a1bed9.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6361/app_12099_PR.apk

**Comment by sethalves** [comment]
to test:
- enable advanced menus
- open running-scripts page
- type "run" into the search field, expand tabs, run "run.js"
- enable "advanced movement"
- try the "run" button that has been added to the toolbar or tablet
- with and without HMD, experiment with walking and running around domains with different terrain and structures
- movement should be as expected, and when in 1st person camera, you shouldn't see any of your head.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6361/HighFidelity-Beta-PR12099-d3084dc0df71a91065683747a89944710f358812.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6361/HighFidelity-Beta-PR12099-d3084dc0df71a91065683747a89944710f358812.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6363/app_12099_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6363/HighFidelity-Beta-PR12099-1e847cf32e3ab92a398d7b6ab3675265fa5b2c2b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6363/HighFidelity-Beta-PR12099-1e847cf32e3ab92a398d7b6ab3675265fa5b2c2b.dmg

**Comment by QAadahlin** [comment]
Currently Testing

**Comment by QAadahlin** [comment]
@conklin94122 Completed Testing for PR#12099. PASSED.

**Pull request merged**